### PR TITLE
feat: MLSStateHandler updates clientEntities directly

### PR DESCRIPTION
**Pull request by aweiss-dev**
Change the implementation to utilise the new observable isMLSVerified on the clientEntity level.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "@peculiar/x509": "1.9.5",
     "@wireapp/avs": "9.4.18",
     "@wireapp/commons": "5.2.1",
-    "@wireapp/core": "42.13.0",
+    "@wireapp/core": "^42.14.0",
```
Comment (aweiss-dev) [inline package.json]:
```suggestion
    "@wireapp/core": "42.14.0",
```

File: server/yarn.lock [hunk 2]
```diff
@@ -26,32 +26,32 @@ __metadata:
   linkType: hard
```
Comment (aweiss-dev) [inline server/yarn.lock:27]: I needed to fix yarn.lock issues. 

Did that by clearing the locks and rebuilding them freshly. Thats why some packages got updates

**Review by przemvs — APPROVED** [review]

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts [hunk 3]
```diff
@@ -140,20 +152,13 @@ export class MLSConversationVerificationStateHandler {
...
-    const {isResultComplete, identities, qualifiedIds} =
-      await this.getAllUserEntitiesInConversation(conversationEntity);
+    const {isResultComplete, qualifiedIds} = await this.updateUserDevices(conversationEntity);
 
     // If the number of userDevicePairs is not equal to the number of identities, our Conversation is not secure
     if (!isResultComplete) {
       return this.degradeConversation(conversationEntity, qualifiedIds);
     }
 
-    // We need to check if identities are valid and not expired
```
Comment (aweiss-dev) [inline src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts:151]: This check is being done in updateUserDevices now

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts [hunk 4]
```diff
@@ -95,31 +97,41 @@ export class MLSConversationVerificationStateHandler {
...
+        }))
+        .reduce((acc, current) => {
+          return {...acc, ...current};
+        }, {});
+      const identities = await this.core.service!.e2eIdentity!.getUserDeviceEntities(
+        conversation.groupId!,
+        deviceUserPairs,
+      );
+      identities.forEach(async identity => {
+        const verified = await this.isCertificateActiveAndValid(identity.certificate);
```
Comment (aweiss-dev) [inline src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts:120]: Verify the certificate, add it to allIdentities only when being valid

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts [hunk 5]
```diff
@@ -95,31 +97,41 @@ export class MLSConversationVerificationStateHandler {
...
+      const devices = userEntity.devices();
+      const deviceUserPairs = devices
+        .map(device => ({
+          [device.id]: userEntity.qualifiedId,
+        }))
+        .reduce((acc, current) => {
+          return {...acc, ...current};
+        }, {});
+      const identities = await this.core.service!.e2eIdentity!.getUserDeviceEntities(
+        conversation.groupId!,
```
Comment (atomrc) [inline src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: I think there is a `isMLSConversation` that will also make sure that this `groupId` is present ;)

Comment (atomrc) [reply src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: have a look at the `ConversationSelector.ts`

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts [hunk 6]
```diff
@@ -95,31 +97,41 @@ export class MLSConversationVerificationStateHandler {
...
+        deviceUserPairs,
+      );
+      identities.forEach(async identity => {
+        const verified = await this.isCertificateActiveAndValid(identity.certificate);
+        if (verified) {
+          const device = devices.find(device => device.id === identity.clientId);
+          /**
+           * ToDo: Change the current implementation of isMLSVerified to be stored in Zustand instead of ko.observable
+           */
+          device?.meta.isMLSVerified?.(true);
```
Comment (atomrc) [inline src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts:126]: This meta is saved to the DB, right?

**Review by atomrc — APPROVED** [review]

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts (hunk 6 above)
Comment (aweiss-dev) [reply src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts:126]: Yes, it should. I added all the entries for that.
The DB entries get initialized with it, and it is also added in the devices toJSON method.

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts (hunk 5 above)
Comment (aweiss-dev) [reply src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: good point. Changed the types and added the guard

**Review by PatrykBuniX — APPROVED** [review]

**Pull request merged**